### PR TITLE
test: add basic pathfinding test cases

### PR DESCRIPTION
**Pull request by jdrueckert**
Extract many tests cases from Terasology/Behaviors#92 to tests on the pathfinding itself.

Most importantly, this allows us to run these tests as _unit tests_ instead of long-running integration tests using MTE.

### CODE REVIEW

File: src/test/java/org/terasology/flexiblepathfinding/FallingJPSTest.java [hunk 1]
```diff
@@ -0,0 +1,184 @@
...
+package org.terasology.flexiblepathfinding;
+
+import org.junit.jupiter.api.Test;
+import org.terasology.flexiblepathfinding.helpers.JPSTestHelper;
+import org.terasology.flexiblepathfinding.helpers.TestPaths;
+import org.terasology.flexiblepathfinding.helpers.TestWorlds;
+import org.terasology.flexiblepathfinding.plugins.basic.FallingPlugin;
+
+/**
+ * Falling movements are purely vertical (down) and require additional horizontal walking movements
```
Comment (skaldarnar) [inline src/test/java/org/terasology/flexiblepathfinding/FallingJPSTest.java]: The term "movement" or "movement type" is introduce one level higher in Behaviors for the actual physical movement input.

```suggestion
 * Falling describes purely vertical (down) paths and does not allow for horizontal steps.
 *
```

File: src/test/java/org/terasology/flexiblepathfinding/FallingJPSTest.java [hunk 2]
```diff
@@ -0,0 +1,184 @@
...
+ * The following test cases only attempt to test that the falling plugin alone does not incorrectly find valid paths for basic movements
```
Comment (skaldarnar) [inline src/test/java/org/terasology/flexiblepathfinding/FallingJPSTest.java:15]: Can you add one positive test case to ensure that as well? Basically, finding a path from a position in the air one block down to a "grounded" block.

File: src/test/java/org/terasology/flexiblepathfinding/LeapingJPSTest.java [hunk 3]
```diff
@@ -0,0 +1,185 @@
...
+import org.junit.jupiter.api.Test;
+import org.terasology.flexiblepathfinding.helpers.JPSTestHelper;
+import org.terasology.flexiblepathfinding.helpers.TestPaths;
+import org.terasology.flexiblepathfinding.helpers.TestWorlds;
+import org.terasology.flexiblepathfinding.plugins.basic.LeapingPlugin;
+
+/**
+ * Leaping movements are purely vertical (up) and require additional horizontal walking movements
+ * The following test cases only attempt to test that the leaping plugin alone does not incorrectly find valid paths for basic movements
+ **/
```
Comment (skaldarnar) [inline src/test/java/org/terasology/flexiblepathfinding/LeapingJPSTest.java]: Same clarification as above, plus the request to add one positive test case.

```suggestion
/**
 * Leaping describes purely vertical (up) paths and does not allow for horizontal steps.
 *
 * The following test cases (mostly) attempt to test that the leaping plugin alone does not incorrectly find valid paths that require horizontal steps.
 **/
```

File: src/test/java/org/terasology/flexiblepathfinding/WalkingJPSTest.java [hunk 4]
```diff
@@ -0,0 +1,180 @@
...
+package org.terasology.flexiblepathfinding;
+
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.provider.Arguments;
+import org.terasology.flexiblepathfinding.helpers.JPSTestHelper;
+import org.terasology.flexiblepathfinding.helpers.TestPaths;
+import org.terasology.flexiblepathfinding.helpers.TestWorlds;
+import org.terasology.flexiblepathfinding.plugins.basic.WalkingPlugin;
+
+public class WalkingJPSTest {
```
Comment (skaldarnar) [inline src/test/java/org/terasology/flexiblepathfinding/WalkingJPSTest.java:17]: Add javadoc similar to to LeapingTests and FallingTests to explain that Walking should find paths with only horizontal steps, and that no vertical steps are allowed?

File: src/test/java/org/terasology/flexiblepathfinding/helpers/TestPaths.java [hunk 5]
```diff
@@ -0,0 +1,152 @@
+// Copyright 2022 The Terasology Foundation
+// SPDX-License-Identifier: Apache-2.0
+
+package org.terasology.flexiblepathfinding.helpers;
+
+public final class TestPaths {
+    public static final String[] singleFlatStepNorthPath = {
```
Comment (skaldarnar) [inline src/test/java/org/terasology/flexiblepathfinding/helpers/TestPaths.java]: Checkstyle wants you to write this as 
```suggestion
    public static final String[] SINGLE_FLAT_STEP_NORTH_PATH= {
```

not sure this is better... maybe we should disable checkstyle for the whole file?

Comment (jdrueckert) [reply src/test/java/org/terasology/flexiblepathfinding/helpers/TestPaths.java]: I'm okay with both

**Review by skaldarnar — APPROVED** [review]

File: src/test/java/org/terasology/flexiblepathfinding/FallingJPSTest.java (hunk 2 above)
Comment (jdrueckert) [reply src/test/java/org/terasology/flexiblepathfinding/FallingJPSTest.java:15]: done

File: src/test/java/org/terasology/flexiblepathfinding/WalkingJPSTest.java (hunk 4 above)
Comment (jdrueckert) [reply src/test/java/org/terasology/flexiblepathfinding/WalkingJPSTest.java:17]: done

**Comment by jdrueckert** [comment]
[FlexibleMovement as a replacement for PathFinding](https://trello.com/c/AGtBwude/189-flexiblemovement-as-a-replacement-for-pathfinding)

**Pull request merged**